XenAdmin/Wizards/PatchingWizard/AutomatedUpdatesBasePage.cs

```diff
@@ -342,20 +343,22 @@ private void WorkerDoWork(object sender, DoWorkEventArgs doWorkEventArgs)
                             var existing = hp.DelayedPlanActions.FirstOrDefault(a => a.GetType() == dpa.GetType());
                             if (existing == null)
                                 hp.DelayedPlanActions.Add(dpa);
+                            if (dpa is RestartHostPlanAction)
+                                hp.RequireDelayedRestartHost = true;
                         }
                     }
 
                     if (hp.DelayedPlanActions.Any(a => a is RestartHostPlanAction))
                     {
-                        if (!Helpers.ElyOrGreater(host) || host.updates_requiring_reboot.Count > 0)
-                            hp.DelayedPlanActions.RemoveAll(a => a is RestartAgentPlanAction);
-                        else
+                        if (Helpers.ElyOrGreater(host) && host.updates_requiring_reboot.Count <= 0 && !hp.RequireDelayedRestartHost)

```

Comment:
I think it's safer to re-resolve the host like in the original code: `host.Connection.TryResolveWithTimeout(new XenRef<Host>(host.opaque_ref)).updates_requiring_reboot.Count`.

Reply:
yes